src/main/java/com/iota/iri/service/tipselection/impl/ConnectedComponentsStartingTipSelector.java

```diff
@@ -76,61 +72,15 @@ public Hash getTip() throws Exception {
         return result;
     }
 
-    private Collection<Hash> findNMostRecentTransactions(Collection<Hash> tips) throws Exception {
-
-        Collection<Hash> result = new HashSet<>(maxTransactions);
-
-        //using a max heap sorted by arrivalTime.
-        Queue<TransactionViewModel> queue = new PriorityQueue<>(maxTransactions,
-                Comparator.comparingLong(a -> (-1) * a.getArrivalTime()));
-        Set<Hash> visited = new HashSet<>(maxTransactions);
-
-        //the heap is initialized with the tips.
-        tips.stream().map(this::fromHash).forEach(queue::add);
-        visited.addAll(tips);
-
-        while (!queue.isEmpty() && result.size() < maxTransactions) {
-            TransactionViewModel current = queue.poll();
-            result.add(current.getHash());
-
-            //add children to priority queue, only if not already chosen.
-            Hash trunkHash = current.getTrunkTransactionHash();
-            Hash branchHash = current.getBranchTransactionHash();
-
-            List<Hash> approvees = Arrays.asList(trunkHash, branchHash);
-            for (Hash approvee: approvees) {
-                if (!visited.contains(approvee)) {
-                    queue.add(TransactionViewModel.fromHash(tangle, approvee));
-                    visited.add(approvee);
-                }
-            }
-        }
-
-        return result;
-    }
-
-    private Hash randomlySelectTipFromLargestConnectedComponent(Collection<Set<Hash>> connectedComponents,
-                                                               Collection<Hash> tips) {
+    private Hash randomlySelectTransactionFromLargestConnectedComponent(Collection<Set<Hash>> connectedComponents) {
         Collection<Hash> largestComponent = connectedComponents.stream()
                 .max(Comparator.comparing(Collection::size))
                 .orElse(Collections.emptySet());
 
-        List<Hash> tipsInLargestComponent = tips.stream()
-                .filter(largestComponent::contains)
-                .collect(Collectors.toList());
-
-        if (tipsInLargestComponent.isEmpty()) {
-            throw new IllegalStateException("no tips found in largest connected component.");
-        }
-
-        return tipsInLargestComponent.get(random.nextInt(tipsInLargestComponent.size()));
```

Comment:
so how do you select a tip in the largest component?
i.e. did you mean to remove the part that intersects w/ tips?